ipaserver/install/custodiainstance.py

```diff
@@ -70,7 +71,14 @@ def uninstall(self):
             'server_keys': self.server_keys,
             'ldap_uri': self.ldap_uri
         })
-        keystore.remove_server_keys()
+        keystore.remove_server_keys_file()
+        try:
+            keystore.remove_server_keys()
+        except (ldap.CONNECT_ERROR, ldap.SERVER_DOWN):
```

Comment:
remove_server_keys() is internally calling remove_keys(), which calls also remove_server_keys_file(), so we end up calling twice in a row the function remove_server_keys_file().

Reply:
Yes, I did this on purpose. It's ok.

Reply:
Sorry, but I don't understand why you need to keep the call to keystore.remove_server_keys_file() on ipaserver/install/custodiainstance.py:74 as remove_keys already calls remove_server_keys_file(). Or am I missing something?

Reply:
``remove_keys`` may fail and I want to make sure that the key file is always removed.

Reply:
Then in this case you can remove the call self.remove_server_keys_file() from ipaserver/secrets/kem.py on line 255.

Reply:
No, we cannot do this. This would change the behavior of ``remove_keys()``. It used to remove keys from LDAP and file system. I like to keep it that way **and** ensure that the keys are always removed from the file system -- no matter how ``remove_keys()`` is failing.

The code is carefully written with defense-in-depths in mind. ``remove_server_keys_file()`` is reentrant and idempotent.

Reply:
Please add a comment explaining your intentions so that we wouldn't get to argue about it again in future when the whole discussion at github is not available.